### PR TITLE
fix(core): 13914 Reset login state if getting 401 from the server

### PR DESCRIPTION
**Pull request by vkozio**
https://kontur.fibery.io/Tasks/Task/Reset-login-state-if-getting-401-from-the-server-13914
full reload after login and logout
fixes for apiclient
removed some unused code

### CODE REVIEW

File: src/core/api_client/tests/authorization.test.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ test('refreshToken called when token is expired', async ({ ctx }) => {
 
   // set private field token with new token
   const apiClientObj = ctx.apiClient as any;
-  apiClientObj.tokenWillExpire = new Date(new Date().getTime() + 1000 * 60 * 4);
+  apiClientObj.tokenExpirationDate = new Date(new Date().getTime() + 1000 * 60 * 4);
```
Comment (Komzpa) [inline src/core/api_client/tests/authorization.test.ts:137]: Can any comment on magic number be added?

File: src/core/auth/client/AuthClient.ts [hunk 2]
```diff
@@ -85,22 +87,24 @@ export class AuthClient {
     const response = await this._apiClient.login(user, password);
     if (response && typeof response === 'object' && 'token' in response) {
       await this.processAuthResponse(response);
+      // reload to init with authenticated config and profile
+      location.reload();
       return true;
```
Comment (IlyaIzr) [inline src/core/auth/client/AuthClient.ts:92]: Do you think expired token can caught user surprised and break it's user flow?

File: src/core/auth/client/AuthClient.ts [hunk 3]
```diff
@@ -85,22 +87,24 @@ export class AuthClient {
...
     }
     return response;
   }
 
   public async checkAuth(): Promise<boolean> {
     try {
-      const response = this._apiClient.getLocalAuthToken(this.onTokenExpired);
+      const response = this._apiClient.getLocalAuthToken(() => this.onTokenExpired());
```
Comment (IlyaIzr) [inline src/core/auth/client/AuthClient.ts:99]: why did you wrap it in arrow function?

Comment (vkozio) [reply src/core/auth/client/AuthClient.ts:99]: to bind this

File: src/core/auth/client/AuthClient.ts (hunk 2 above)
Comment (vkozio) [reply src/core/auth/client/AuthClient.ts:92]: yes but it's expected, smoother experience will be implemented in other tasks

**Review by IlyaIzr — APPROVED** [review]
questionable stuff was discussed in threads

**Pull request merged**